src/pyscheduling/JS/JobShop.py

```diff
@@ -457,6 +938,115 @@ def __init__(self, instance: JobShopInstance = None, machines: list[Machine] = N
         else:
             self.machines = machines
         self.objective_value = 0
+        self.graph = graph
+        self.job_schedule = {j: Job(j, 0, 0) for j in range(self.instance.n)}
+
+    def create_solution_graph(self, invert_weights = True):
```

Comment:
Here we can create the disjunctive arcs from the solution (even when it's a partial one).

If you need very specific methods for the graph (adding edges or updating weights), these can be implemented in the graph class. Otherwise, completion times, objective, is_feasible ... should be in the solution.